packages/amplify-gen1-codegen-auth-adapter/API.md

```diff
@@ -0,0 +1,21 @@
+## API Report File for "@aws-amplify/amplify-gen1-codegen-auth-adapter"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { AuthDefinition } from '@aws-amplify/amplify-gen2-codegen';
+import { UserPoolType } from '@aws-sdk/client-cognito-identity-provider';
```

Comment:
Didn't we decide to not use SDK types in APIs ?

Reply:
We discussed not using them in the codegen package so that the codegen package can stay independent. The purpose of the `amplify-gen1-codegen-{auth,storage}` packages are to reduce descriptions of AWS resources down to the inputs required for codegen. This allows codegen to specify its own types and API, while creating a simple way to adhere to that API using common SDK outputs. 

I broke out these separate adapter packages so that directly leveraging SDK types would be more feasible.

Reply:
The idea isn't bad.

However it has two risks:
1. It's easier to enforce single rule, i.e. "no sdk types" rather than "in this package sdk types are ok". Perhaps this script could be enhanced to enforce it https://github.com/aws-amplify/amplify-cli/blob/dev/scripts/verify-extract-api.sh (but it's some effort to get it right that we might want to avoid).
2. If top level codegen ever re-exports/uses these types they'll transitively get exposed there.


Perhaps middle ground is to continue with this approach but have a release blocker to inspect this later.